core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -141,6 +143,9 @@ private[spark] class TaskSetManager(
   // Task index, start and finish time for each task attempt (indexed by task ID)
   val taskInfos = new HashMap[Long, TaskInfo]
 
+  // Use a MedianHeap to record durations of successful tasks when speculation is enabled.
```

Comment:
can you expand on this a bit to something like:

// Use a MedianHeap to record durations of successful tasks so we know when to launch speculative tasks.
// This is only used when speculation is enabled, to avoid the overhead of inserting into the heap when the
// heap won't be used.